### ISSUE TITLE
augmentation.mix: RandomTransplantation without a mask raises Python's IndexError/list.index ValueError

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4777]
`RandomTransplantation` and `RandomTransplantation3D` need a mask to draw their parameters. When the call does not provide one, the error comes from Python's tuple indexing or `list.index`, not from kornia, and names neither the mask nor `data_keys`.

## Repro

```python
import torch, kornia, kornia.augmentation as K
print(kornia.__version__, torch.__version__)
for cls, spatial in ((K.RandomTransplantation, (6, 8)), (K.RandomTransplantation3D, (3, 6, 8))):
    image = torch.rand(3, 3, *spatial)
    for call in (lambda a: a(image), lambda a: a(image, data_keys=["input"])):
        try:
            call(cls(p=1.0))
        except Exception as e:
            print(cls.__name__, type(e).__name__, e)
```

## Observed (kornia 0.9.0rc1 @ 57cd68e6f, torch 2.14.0, CPU)

```
RandomTransplantation IndexError tuple index out of range
RandomTransplantation ValueError <DataKey.MASK: 1> is not in list
RandomTransplantation3D IndexError tuple index out of range
RandomTransplantation3D ValueError <DataKey.MASK: 1> is not in list
```

The first call uses the default `data_keys` (`["input", "mask"]`) with only the image passed; the second names no mask key.

## Expected

A kornia error (for example `ValueError`) raised before any indexing that says a `"mask"` input is required to draw the transplant parameters, and, in the first case, that the number of inputs does not match `data_keys`.

## Why this is a bug

The mask requirement is documented, but not validated: the failure surfaces wherever the missing value is first indexed. Fix direction: check `DataKey.MASK in keys` and `len(input) == len(keys)` at the top of `forward` when `params` has to be drawn.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).
